examples/reverser/server.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/ServiceWeaver/weaver"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+type server struct {
+	weaver.Implements[weaver.Main]
+	reverser Reverser
+	lis      net.Listener
+}
+
+func (s *server) Init(context.Context) error {
+	// Get a client to the Reverser component.
+	var err error
+	s.reverser, err = weaver.Get[Reverser](s)
+	if err != nil {
+		return err
+	}
+
+	// Get a network listener.
+	opts := weaver.ListenerOptions{LocalAddress: *address}
+	s.lis, err = s.Listener("reverser", opts)
```

Comment:
If listener is fetched in serve() below, the test will become simpler (it won't need to create a listener itself).